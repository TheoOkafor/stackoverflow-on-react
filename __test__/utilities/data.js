const data = {
  answer: {
    accepted: null,
    body: 'What?',
    comments: [
      {
        answerid: 17,
        body: 'Jonzing',
        id: 30,
        questionid: 4,
        timesubmitted: '2018-11-21T12:35:08.000Z',
        userid: 4,
        username: 'teddy01'
      },
    ],
    downvotes: 0,
    id: 74,
    questionid: 38,
    timesubmitted: '2018-11-21T18:20:11.000Z',
    upvotes: 0,
    userid: 4,
    username: 'teddy01',
    votesCount: 0,
  },
  comment: {
    answerid: 17,
    body: 'Jonzing',
    id: 30,
    questionid: 4,
    timesubmitted: '2018-11-21T12:35:08.000Z',
    userid: 4,
    username: 'teddy01'
  },
  sampleAnswer: {
    data: {
      body: 'Jonzing',
      questionid: 4,
      timesubmitted: '2018-11-21T12:35:08.000Z',
      userid: 4,
      username: 'teddy01'
    },
  },
  question: {
    data: {
      answers: [
        {
          accepted: false,
          body: 'Lorem ipsum dolor sit amet, ',
          comments: [],
          downvotes: 0,
          id: 4,
          questionid: 4,
          timesubmitted: '2018-08-22T11:13:09.000Z',
          upvotes: 1,
          userid: 3,
          username: 'chanpion',
          votesCount: 1
        },
        {
          accepted: false,
          body: 'Lorem ipsum dolor sit amet, ',
          comments: [],
          downvotes: 0,
          id: 8,
          questionid: 4,
          timesubmitted: '2018-08-22T11:13:09.000Z',
          upvotes: 0,
          userid: 3,
          username: 'chanpion',
          votesCount: 0
        },
        {
          accepted: false,
          body: 'Lorem ipsum dolor sit amet, ',
          comments: [],
          downvotes: 0,
          id: 12,
          questionid: 4,
          timesubmitted: '2018-08-22T11:13:09.000Z',
          upvotes: 0,
          userid: 3,
          username: 'chanpion',
          votesCount: 0
        },
        {
          accepted: false,
          body: 'Lorem ipsum dolor sit amet, ',
          comments: [],
          downvotes: 0,
          id: 16,
          questionid: 4,
          timesubmitted: '2018-08-22T11:13:09.000Z',
          upvotes: 0,
          userid: 3,
          username: 'chanpion',
          votesCount: 0
        },
        {
          accepted: false,
          body: 'Lorem ipsum dolor sit amet, ',
          comments: [{
            answerid: 17,
            body: 'Jonzing',
            id: 30,
            questionid: 4,
            timesubmitted: '2018-11-21T12:35:08.000Z',
            userid: 4,
            username: 'teddy01'
          }],
          downvotes: 0,
          id: 17,
          questionid: 4,
          timesubmitted: '2018-08-22T11:13:09.000Z',
          upvotes: 0,
          userid: 3,
          username: 'chanpion',
          votesCount: 0
        },
        {
          accepted: null,
          body: 'worddy',
          comments: [],
          downvotes: 0,
          id: 67,
          questionid: 4,
          timesubmitted: '2018-11-21T12:23:12.000Z',
          upvotes: 0,
          userid: 4,
          username: 'teddy01',
          votesCount: 0
        }
      ],
      body: 'Lorem ipsum dolor sit amet, ',
      id: 4,
      timesubmitted: '2018-08-22T11:13:09.000Z',
      title: 'How can one learn it all so fast?',
      userid: 3,
      username: 'chanpion'
    },
    message: 'Question 4 found',
    statusCode: 200
  },
  questions: [
    {
      id: 1,
      title: 'why do yo shake?',
      body: 'Lorem ipsum dolor sit amet, ',
      userid: 3,
      username: 'chanpion',
      answers: [],
    },
    {
      id: 2,
      title: 'why do yo shake?',
      body: 'Lorem ipsum dolor sit amet, ',
      userid: 3,
      username: 'chanpion',
      answers: [],
    },
    {
      id: 3,
      title: 'why do yo shake?',
      body: 'Lorem ipsum dolor sit amet, ',
      userid: 3,
      username: 'chanpion',
      answers: [],
    },
  ],
  newQuestion: {
    id: 1,
    title: 'why do yo shake?',
    body: 'Lorem ipsum dolor sit amet, ',
    userid: 3,
    username: 'chanpion',
    answers: [],
  },
  payload: {
    username: 'Xtreme',
    'x-access-token': 'token',
  },
  userData: {
    firstName: 'Xtreme',
    lastName: 'Prodigy',
    username: 'Xtreme',
    password: 'password',
    confirmpassword: 'password',
  },
};

export default data;
