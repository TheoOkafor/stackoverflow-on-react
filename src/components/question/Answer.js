import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import AnswerOptions from './answer/AnswerOptions';
import AcceptAnswer from './answer/AcceptAnswer';
import Comment from './answer/Comment';

const Answer = (props) => {
  return (
    <div className="answer-card">
      <div>
        <h4 className="person-answer left">
          <Link to={`/users/${props.answer.userid}`}
            className="inherit">{props.answer.username}</Link>
          <br />
          <small>Answered:
            <span>{new Date(props.answer.timesubmitted).toDateString()}
            </span></small>
        </h4>

        <div className="clear-fix" />
        <p className="answer" id={`"ans-${props.answer.id}"`}>
          {props.answer.body}
        </p>

        <AcceptAnswer
          answer={props.answer}
          index={props.index}
          hasAccepted={props.hasAccepted}
          username = {props.question.username}
          handleAccept={props.handleAccept} />
        <AnswerOptions answer={props.answer} index={props.index} />

        <Comment answer={props.answer} />
      </div>
    </div>
  );
};

Answer.propTypes = {
  answer: PropTypes.object,
  index: PropTypes.number,
  hasAccepted: PropTypes.bool,
  question: PropTypes.object,
  handleAccept: PropTypes.func,
};

export default Answer;
