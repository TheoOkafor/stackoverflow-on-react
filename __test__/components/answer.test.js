import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Answer from '../../src/components/question/Answer';
import store from '../../src/store';
import data from '../utilities/data';

test('Test that the AnswerOptions displays', () => {
  const component = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <Answer
          answer={data.answer}
          question={data.question} />
      </MemoryRouter>
    </Provider>
  );
  expect(component).toMatchSnapshot();
});
