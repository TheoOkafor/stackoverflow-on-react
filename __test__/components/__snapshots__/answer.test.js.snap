// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test that the AnswerOptions displays 1`] = `
<div
  className="answer-card"
>
  <div>
    <h4
      className="person-answer left"
    >
      <a
        className="inherit"
        href="/users/4"
        onClick={[Function]}
      >
        teddy01
      </a>
      <br />
      <small>
        Answered:
        <span>
          Wed Nov 21 2018
        </span>
      </small>
    </h4>
    <div
      className="clear-fix"
    />
    <p
      className="answer"
      id="\\"ans-74\\""
    >
      What?
    </p>
    <div />
    <div
      className="answer-options"
    >
      <div
        className="message-display"
      >
        <p
          id="vote-mssg-74"
        />
        <div
          className="loader-xs"
          id={74}
        />
      </div>
      <span
        className="upvote-display"
      >
        0
      </span>
      <button
        className="btn primary-o"
        data-testid="upvote"
        id={74}
        onClick={[Function]}
        value="upvote"
      >
        Upvote
      </button>
      <span
        className="vote-display"
      >
        0
      </span>
      <button
        className="btn link gray"
        data-testid="downvote"
        id={74}
        onClick={[Function]}
        value="downvote"
      >
        Downvote
      </button>
      <button
        className="btn link"
      >
        Comment
      </button>
    </div>
    <div
      className="comments"
    >
      <p
        className="comment-display"
      >
        <small>
          Jonzing
          <span
            className="right"
          >
            <a
              className="inherit"
              href="/users/4"
              onClick={[Function]}
            >
              <b>
                teddy01
              </b>
            </a>
            (
            Wed Nov 21 2018
            )
          </span>
        </small>
      </p>
      <form
        className="comment-form"
      >
        <div
          className="message-display"
        >
          <p
            id="server-message-74"
          />
          <div
            className="loader-xs"
            id="comment-74-loader"
          />
        </div>
        <textarea
          id="comment-box-74"
          placeholder="Comment on this answer"
        />
        <button
          className="btn primary"
          id="comment-74"
        >
          Post Comment
        </button>
        <input
          type="reset"
          value="Cancel"
        />
      </form>
    </div>
  </div>
</div>
`;
